opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/PlatformResources.java

```
@@ -33,113 +37,469 @@
+		if(statuses == null)
+		{
+			return new ArrayList<>();
+		}
+		List<ApiPlatformStatus> ret = new ArrayList<>();
+		for(PlatformStatus status : statuses)
 		{
-			return ApiHttpUtil.createResponse(platformDAO.getPlatformStatus(netlistId));
+			ApiPlatformStatus ps = new ApiPlatformStatus();
+			if(status.getPlatformId() != null)
+			{
+				ps.setPlatformId(status.getPlatformId().getValue());
+			}
+			else
+			{
+				ps.setPlatformId(DbKey.NullKey.getValue());
+			}
+			ps.setAnnotation(status.getAnnotation());
+			ps.setLastContact(status.getLastContactTime());
+			ps.setLastError(status.getLastErrorTime());
+			ps.setLastMessage(status.getLastMessageTime());
+			if (dbIo != null && !status.getLastScheduleEntryStatusId().isNull()
+					&& (status.getLastRoutingSpecName() == null || status.getLastRoutingSpecName().isEmpty()))
+			{
+				try (ScheduleEntryDAI dai = dbIo.makeScheduleEntryDAO())
+				{
+					ScheduleEntry scheduleEntry = dai.readScheduleEntry(status.getLastScheduleEntryStatusId());
+					long routingId = scheduleEntry.getRoutingSpecId().getValue();
+					RoutingSpec rs = new RoutingSpec();
+					rs.setId(DbKey.createDbKey(routingId));
+					dbIo.readRoutingSpec(rs);
+					ps.setRoutingSpecName(rs.getName());
+				}
+				catch (DatabaseException | DbIoException ex)
+				{
+					// Ignore
+				}
+			}
+			else
+			{
+				ps.setRoutingSpecName(status.getLastRoutingSpecName());
+			}
+			ps.setLastRoutingExecId(status.getLastScheduleEntryStatusId().getValue());
+			if (dbIo != null && status.getPlatformId() != null)
+			{
+				try
+				{
+					Platform pl = new Platform();
+					pl.setId(status.getPlatformId());
+					dbIo.readPlatform(pl);
+					if (pl.getSite() != null && pl.getSite().getId() != null)
+					{
+						ps.setSiteId(pl.getSite().getId().getValue());
+					}
+					if (pl.getDisplayName() != null)
+					{
+						ps.setPlatformName(pl.getSite().getUniqueName());
+					}
+				}
+				catch (DatabaseException ex)
+				{
+					// Ignore
```

never ignore exceptions

---

Fixed